envoy/http/filter.h

```diff
@@ -626,6 +641,33 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
   virtual absl::optional<absl::string_view> upstreamOverrideHost() const PURE;
 };
 
+/**
+ * Hepler used to set and get stream filter names. This is to avoid having to implement
+ * these method for every filter.
+ */
+class StreamFilterNameHepler {
+public:
+  /**
+   * Get custom filter name of current filter instance. This name typically should be the
+   * name of HttpFilter proto configuration.
+   */
+  absl::string_view customName() const { return custom_name_; }
+  /**
+   * Get filter name of current filter instance.
+   */
+  absl::string_view filterName() const { return filter_name_; }
+
+  void setCustomName(absl::string_view view) { custom_name_ = std::string(view); }
+  void setFilterName(absl::string_view view) { filter_name_ = std::string(view); }
+
+protected:
+  // TODO(wbpcode): typically, the custom name and filter name have longer lifetime than the
+  // stream filter instances. We can simply keep string view here to avoid unnecessary copy
+  // and memory allocation. Currently, new strings are still always created for safety.
+  std::string custom_name_;
+  std::string filter_name_;
```

Comment:
not sure it is worth or not, in case of the filter doesn't have custom name, there will be a duplicated of the same name, maybe using the std::shared_ptr to able to get rid of this duplicated name.

Reply:
I will try to make these names to be string_view finally. And I think it's unnecessary to introduce a shared_ptr for single string. 🤔